### PR TITLE
Fixing linter errors

### LINKED ISSUE #4537: Linters are not running
It looks like during the introduction of Vue (or some other event) we stopped applying the Open Library stylelint configuration file. This leads to inconsistent, hard to read code and adds unnecessary friction during the code review process.

Running 
```
npx stylelint ./**/*.less --config .stylelintrc.json
```
triggers various errors.

### Proposal & Constraints
- [x] Make sure the  .stylelintrc.json is applied in the existing lint rules.
- [x] Run npx stylelint ./**/*.less --config .stylelintrc.json --fix
- [x] Any errors that require manual intervention should be fixed
- [x] Spaces between properties and values and !important are fixed and enforced e.g. `background-color:@light-yellow!important; ` errors, and `background-color: @light-yellow !important; ` does not.
 - [x] Instances of #eee are replaced with @lightest-grey variable
- [x] We do not use      // stylelint-disable-next-line function-url-quotes 
- [x] donateModal is replaced with a BEM class

### Stakeholders
@cdrini @mekarpeles

### PR DESCRIPTION
**Pull request by navneetsaluja**
Closes #4537 

refactor 
### Technical

will edit after review. 

### Stakeholders

@jdlrobson

### CODE REVIEW

**Comment by navneetsaluja** [comment]
@jdlrobson Even after extending the `.stylelintrc.json` to the css file, it still allows specificity of (0, 0, 1) in `donate.less`, any reason why? 
Also, how do I check if `donate.less` is working properly? Can't get the donate button to appear locally. I copied a work (which has the donate button) from the prod instance into local but it does not show the donate button locally. 

List of errors after running `npx stylelint ./**/*.less --config .stylelintrc.json --fix` here: 
https://gist.github.com/navneetsaluja/b81ab0e4eb295446332db284a418ff74

Working on the specific no-descending-specificity cases right now.

**Comment by jdlrobson** [comment]
it's okay to disable the specificity rules (e.g.  no-descending-specificity) for existing code on a per line / block basis by adding:
```
/* stylelint-disable selector-max-specificity */
```

We should look to manually fix the other ones though while we are here which should be relatively straightforward.

The max-specificity issues can be fixed later as they are a little riskier to fix.

**Comment by navneetsaluja** [comment]
@jdlrobson Gotcha, I'll see which of the `no-descending-specificity` cases I can fix and disable the rest.

**Comment by navneetsaluja** [comment]
@jdlrobson Please take a look :)

File: static/css/components/buttonCta.less [hunk 1]
```diff
@@ -31,7 +31,8 @@ a.cta-btn {
   }
   // stylelint-disable-next-line selector-max-specificity
   .btn-icon.read-aloud {
-    background-image: url('/static/images/icons/read aloud.svg');
+    // stylelint-disable-next-line function-url-quotes
```
Comment (jdlrobson) [inline static/css/components/buttonCta.less:34]: I think this disabling rule is unnecessary. We just need to use %20 to encode the space in the URL

File: static/css/components/buttonCta.less [hunk 2]
```diff
@@ -65,13 +66,15 @@ a.cta-btn {
     background: lighten(desaturate(@link-blue, 56%), 67%);
   }
   &--vanilla {
```
Comment (jdlrobson) [inline static/css/components/buttonCta.less:68]: That duplicate border rule would have surely confused someone at some point.

File: static/css/components/carousel--js.less [hunk 3]
```diff
@@ -16,7 +16,6 @@
     width: auto;
     height: auto;
 
-    &:before,
     &:before {
```
Comment (jdlrobson) [inline static/css/components/carousel--js.less:19]: How odd! I wonder what happened there.

File: static/css/components/donate.less [hunk 4]
```diff
@@ -47,7 +34,19 @@ h2.donation__override {
...
   }
-  note {
+  &note {
     color: @light-grey;
   }
-}
\ No newline at end of file
+}
+
+div.donateModal {
```
Comment (jdlrobson) [inline static/css/components/donate.less:42]: we can probably drop the div. here. This doesn't look it needs to be so specific. This also has me wondering why this is not using a BEM class like the others. e.g. donate__model

File: static/css/components/preview.less [hunk 5]
```diff
@@ -4,8 +4,8 @@
 
 .book-preview {
   .iframe-container {
-    border-top: 1px solid #eee;
-    border-bottom: 1px solid #eee;
+    border-top: 1px solid rgb(238, 238, 238);
```
Comment (jdlrobson) [inline static/css/components/preview.less:7]: #eee is equivalent to hsl(0, 0%, 93.3%); which is the variable #4550 `@lightest-grey` so please use the variable here.

File: static/css/components/work.less [hunk 6]
```diff
@@ -47,7 +48,7 @@ div.editionAbout {
 }
 
 .work-subtitle{
-  margin-top:0px;
+  margin-top:0;
```
Comment (jdlrobson) [inline static/css/components/work.less:51]: I was surprised stylelint didnt complain about the space here. Is there a rule we could enable to catch this issue and have it fixed?

File: static/css/components/work.less [hunk 7]
```diff
@@ -180,17 +181,8 @@ a.section-anchor {
...
-  margin-bottom: 0;
-}
-
-.work-subtitle{
-  margin-top:0px;
-}
-
 .book-subtitle{
-  margin-bottom:0px;
+  margin-bottom:0;
```
Comment (jdlrobson) [inline static/css/components/work.less:185]: See above about the lack of a space and lack of enforcement for that issue.

File: static/css/components/merge-form.less [hunk 8]
```diff
@@ -4,103 +4,103 @@
...
-      li {
-        color: @grey!important;
-        font-size: .75em;
-      }
-      label {
-        display: block;
-      }
+  }
+  .master {
+    background-color: @light-yellow!important;
```
Comment (jdlrobson) [inline static/css/components/merge-form.less:29]: there is no space here and there should be

**Review by jdlrobson — APPROVED** [review]
This is wonderful to see and has made my morning :)

While reviewing I did notice a few things, but all of those can be addressed in follow ups. It's wonderful to see the code standard improve in our CSS. Thank you!

**Comment by navneetsaluja** [comment]
@jdlrobson I'm confused, why was the PR merged even when there were unresolved errors in the code? Can I still commit to this PR and you'll merge the commits individually or do I open a new follow-up PR? 
About the lack of spaces, there are two stylelint rules we could use, please take a look at them: [declaration-bang-space-before](https://stylelint.io/user-guide/rules/declaration-bang-space-before), [declaration-colon-space-after](https://stylelint.io/user-guide/rules/declaration-colon-space-after)

**Comment by jdlrobson** [comment]
@navneetsaluja I didn't consider any of the issues blocking and for bigger patches like this there's always a risk of merge conflicts and I didn't want to lose the momentum. New pull requests for these outstanding issues would be perfect.

**Comment by navneetsaluja** [comment]
@jdlrobson Will keep this in mind next time, thank you for clarifying. Follow up PR is here #4554.

**Pull request merged**